danger/src/PullRequestManager.ts

```diff
@@ -0,0 +1,165 @@
+    if (changelogTag) {
+      changelogTag
+        .split('\n')
+        .slice(1)
+        .map(line => line.trim())
+        .filter(line => line.length > 0)
+        .forEach(line => {
+          const tags = this.parseTagsFromLine(line);
+          if (!tags) {
+            warn(`Couldn't parse line: ${line}.`);
+            return;
+          }
+
+          changelogEntries[tags.packageName] = {
+            type: tags.type,
+            message: line.replace(/\[.*\]/, '').trim(),
+          };
+        });
+    }
+
+    return changelogEntries;
+  }
+
+  async createOrUpdateRPAsync(
+    missingEntries: { packageName: string; content: string }[]
+  ): Promise<PullRequest | null> {
+    const dangerHeadRef = `@danger/add-missing-changelog-to-${this.pullRequest.number}`;
+    const dangerBaseRef = this.pullRequest.head.ref;
+
+    const fileMap = missingEntries.reduce(
+      (prev, current) => ({
+        ...prev,
+        [getPackageChangelogRelativePath(current.packageName)]: current.content,
+      }),
+      {}
+    );
+
+    await this.githubApi.createOrUpdateBranchFromFileMap(fileMap, {
+      baseBranchName: dangerBaseRef,
+      branchName: dangerHeadRef,
+      message: `${dangerTags} ${dangerMessage}`,
+    });
+
+    const prs = await this.githubApi.getOpenPRs({
+      fromBranch: dangerHeadRef,
+      toBranch: dangerBaseRef,
+    });
+
+    if (prs.length > 1) {
+      warn("Couldn't find the correct pull request. Too many open ones.");
+      return null;
+    }
+
+    if (prs.length === 1) {
+      return prs[0];
+    }
+
+    return this.githubApi.openPR({
+      fromBranch: dangerHeadRef,
+      toBranch: dangerBaseRef,
+      title: `${dangerTags} ${dangerMessage} to #${this.pullRequest.number}`,
+      body: `${dangerMessage} to #${this.pullRequest.number}`,
+    });
+  }
+
+  private parseTagsFromLine(
+    line: string
+  ): { packageName: string | typeof DEFAULT_CHANGELOG_ENTRY_KEY; type: ChangelogEntryType } | null {
+    const result: {
+      packageName: string | typeof DEFAULT_CHANGELOG_ENTRY_KEY;
+      type: ChangelogEntryType;
+    } = {
+      type: DEFAULT_ENTRY_TYPE,
+      packageName: DEFAULT_CHANGELOG_ENTRY_KEY,
+    };
+
+    const tags = line.match(/\[[^\]]*\]/g);
+    if (!tags) {
+      return result;
+    }
+    // We currently support only two tags - packageName and type.
+    if (tags.length > 2) {
+      return null;
+    }
+
+    for (const tag of tags) {
+      switch (true) {
+        case /\[[\s-_]*(bug)?[\s-_]*fix[\s-_]*\]/i.test(tag):
+          result.type = ChangelogEntryType.BUG_FIXES;
+          break;
+        case /\[[\s-_]*(new)?[\s-_]*feature(s)?[\s-_]*\]/i.test(tag):
+          result.type = ChangelogEntryType.NEW_FEATURES;
+          break;
+        case /\[[\s-_]*breaking[\s-_]*(change)?[\s-_]*\]/i.test(tag):
+          result.type = ChangelogEntryType.BREAKING_CHANGES;
+          break;
+        default:
+          result['packageName'] = tag.replace(/\[|\]/g, '').trim();
```

Comment:
Yes, It would be required. I can't add `expo` prefix automatically cause we have packages witch starts with `unimodules` (Although, we only have two prefixes so maybe we can add the `expo` one automatically if the string doesn't start with`unimodules`? ).